sdk/managednetworkfabric/azure-resourcemanager-managednetworkfabric/src/test/java/com/azure/resourcemanager/managednetworkfabric/ManagedNetworkFabricManagerTest.java

```diff
@@ -0,0 +1,196 @@
+import com.azure.resourcemanager.managednetworkfabric.models.IpGroupProperties;
+import com.azure.resourcemanager.managednetworkfabric.models.IpMatchCondition;
+import com.azure.resourcemanager.managednetworkfabric.models.Layer4Protocol;
+import com.azure.resourcemanager.managednetworkfabric.models.PortGroupProperties;
+import com.azure.resourcemanager.managednetworkfabric.models.PortType;
+import com.azure.resourcemanager.managednetworkfabric.models.PrefixType;
+import com.azure.resourcemanager.managednetworkfabric.models.SourceDestinationType;
+import com.azure.resourcemanager.managednetworkfabric.models.VlanGroupProperties;
+import com.azure.resourcemanager.managednetworkfabric.models.VlanMatchCondition;
+import com.azure.resourcemanager.resources.ResourceManager;
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Random;
+
+public class ManagedNetworkFabricManagerTest extends TestBase {
+    private static final Random RANDOM = new Random();
+    private static final Region REGION = Region.US_EAST;
+    private String resourceGroupName = "rg" + randomPadding();
+    private ManagedNetworkFabricManager managedNetworkFabricManager;
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        managedNetworkFabricManager = ManagedNetworkFabricManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        resourceManager = ResourceManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        String registrationState = resourceManager.providers()
+            .getByName("Microsoft.ManagedNetworkFabric").registrationState();
+        if ("NotRegistered".equalsIgnoreCase(registrationState)
+            || "Unregistered".equalsIgnoreCase(registrationState)) {
+            resourceManager.providers().register("Microsoft.ManagedNetworkFabric");
+            registrationState = resourceManager.providers()
+                .getByName("Microsoft.ManagedNetworkFabric").registrationState();
+            while (!"Registered".equalsIgnoreCase(registrationState)) {
+                ResourceManagerUtils.sleep(Duration.ofSeconds(10L));
+                registrationState = resourceManager.providers()
+                    .getByName("Microsoft.ManagedNetworkFabric").registrationState();
+            }
+        }
```

Comment:
I don't think you need to code it there. Just do a register would be OK. And we probably still need to track what namespace we registered on Test subscription.

Reply:
An exception such as `The subscription registration is in 'Unregistered' state. The subscription must be registered to use namespace 'Microsoft.ManagedNetworkFabric'` occurred during the pipeline live test, because I can not register the provider on test subscription through the `azure portal` , so added this code snippet. Now this `Microsoft.ManagedNetworkFabric` has been registered, there will be no problem removed this code snippet.

Reply:
Got it. That's fine. I wasn't aware that the AAD via script actually have the permission to register namespace (ideally it should be one with low permission limited to the resource group...).

In future, maybe just tell us so we do it manually (or maybe have a script to do similar).